### PR TITLE
Documentation: add build and upload with CI

### LINKED ISSUE #4900: Automatic run of Doxygen on pull-requests
Github has announced yesterday a change in Github actions:
https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

I think that would allows us, now, to run the Doxygen compilation automatically on pull-requests, using Github actions.

@maxGimeno I think you can add that in your todo-list.

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Add a workflow using the github actions to build and upload the doc for each new pull_request

## Release Management

* Issue(s) solved (if any): fix #4900

### CODE REVIEW

**Comment by sloriot** [comment]
Can't we use something like [this](https://github.com/Khan/pull-request-comment-trigger) to trigger the build of the doc with a keyword manually added? The link to the doc could also automatically added as a comment.

**Comment by sloriot** [comment]
Note that as demonstrated by Jackson [here](https://jacksoncampolattaro.github.io/cgal-octree-doc/cmake-build-debug/doc_output/Octree/index.html), we could store the doc directly on `cgal.github.io` if we want to save some space on cgal.gf.com for example.

**Comment by maxGimeno** [comment]
I would prefer that rather than storing the doc on my space on cgal.geometryfactory.com

**Comment by maxGimeno** [comment]
This PR is now ready. Secrets will have to be added to CGAL/cgal to make it work. 
Every time a PR is opened against CGAL/cgal:master, the documentation from the associated branch is built, and added to https://cgal.github.io/index.html. 
Every time such a PR is closed, for any reason, the associated doc is removed from the github.io and its index.

File: .github/workflows/delete_doc.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+         git config --global user.name "Maxime Gimeno"
+         git clone git@github.com:CGAL/cgal.github.io.git
+         PR_NUMBER=$(python -c "import json; import os; y = json.load(open(os.environ['GITHUB_EVENT_PATH'])); print(y[\"number\"])")
+         cd cgal.github.io/
+         while IFS= read -r line
+         do
+            if [[ $line != *" ${PR_NUMBER}."* ]]; then
+              echo "$line" >> tmp.html
+            fi
+         done < "index.html"
```
Comment (lrineau) [inline .github/workflows/delete_doc.yml:44]:
```suggestion
         egrep -v " ${PR_NUMBER}\." index.html > tmp.html
         mv tmp.html index.html
```

File: .github/workflows/build_doc.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        cp -r ./doc_output/* ../cgal.github.io/${PR_NUMBER}/
+        cd ../cgal.github.io
+        INDEX_EXISTS=""
+        while IFS= read -r line
+        do
+           if [[ $line == *" ${PR_NUMBER}."* ]]; then
+             INDEX_EXISTS="TRUE"
+           fi
+        done < "index.html"
+        if [ -z "$INDEX_EXISTS" ]; then
```
Comment (lrineau) [inline .github/workflows/build_doc.yml:60]:
```suggestion

        if ! egrep -q " ${PR_NUMBER}\."; then 
```

**Pull request merged**